kotlin-grpc/src/main/kotlin/CoroutineServerInterceptor.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import com.linecorp.armeria.server.grpc.AsyncServerInterceptor
+import io.grpc.Metadata
+import io.grpc.ServerCall
+import io.grpc.ServerCallHandler
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.future.future
+import java.util.concurrent.CompletableFuture
+
+/**
+ * A [ServerInterceptor][io.grpc.ServerInterceptor] that is able to asynchronously execute the interceptor without blocking the
+ * caller thread.
+ * For example:
+ * ```kotlin
+ * class AuthServerInterceptor : CoroutineServerInterceptor {
+ *     override suspend fun <ReqT, RespT> suspendedInterceptCall(
+ *             call: ServerCall<ReqT, RespT>,
+ *             headers: Metadata,
+ *             next: ServerCallHandler<ReqT, RespT>
+ *     ): ServerCall.Listener<ReqT> = suspendCoroutine {
+ *         val future = authorizer.authorize(ServiceRequestContext.current(), headers)
+ *         future.whenComplete { result, _ ->
+ *             if (result) {
+ *                 next.startCall(call, headers)
+ *             } else {
+ *                 throw AnticipatedException("Invalid access")
+ *             }
+ *         }
+ *     }
```

Comment:
We could simply use `.await()` to convert a `CompletableFure`.
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-jdk8/kotlinx.coroutines.future/await.html
```kt
val result = authorizer.authorize(ServiceRequestContext.current(), headers).await()
if (result) {
    next.startCall(call, headers)
} else {
    throw AnticipatedException("Invalid access")
}
```

Reply:
In 47db28e, I applied your suggestion 🙂